php/query_utils.php

```diff
@@ -462,6 +462,139 @@ function get_associated_survey_instances(
+        //         1 |                  NULL |                    NULL |                NULL
+        //         2 |                     1 |                    NULL |                NULL
+        //         3 |                  NULL |                    NULL |                   1
+        //         4 |                  NULL |                       1 |                   3
+        //         5 |                     2 |                       2 |                NULL
+        //         6 |                     1 |                    NULL |                NULL
+        //
+        // in the table above, surveys with survey_id 2,6 are returned for admin with user_id turner32
+        $col_name = $roles[$t_name];
+        $sql =
+            "SELECT survey_id FROM $t_name AS t " .
+            "JOIN surveys AS s " .
+            "ON s.$col_name = t.id " .
+            "WHERE t.user_id = :user_id AND s.term LIKE :term";
+        $bound = [
+            "user_id" => $user_id,
+            "term" => $term
+        ];
+        $res = do_select_query($sql, $bound, $conn);
+        foreach ($res->result as $x) {
+            $ret[] = $x["survey_id"];
+        }
+
+        $cond = "";
+        //if instructor we check if ta choice is also null
+        if ($t_name == "course_survey_choices") {
+            $cond = "AND ta_survey_choice_id IS NULL";
+
+            //if admin we check both instructor and ta choice are also null
+        } elseif ($t_name == "dept_survey_choices") {
+            $cond =
+                "AND course_survey_choice_id IS NULL AND ta_survey_choice_id IS NULL ";
+        }
+        $col_name = $roles[$t_name];
+        // get all surveys where the the choice table of their role
+        // and below their table is null
+        // e.g All surveys with course_survey_choice_id = NULL and
+        //         course_survey_choice_id = NULL are returned for all instructors
+        // survey_id | dept_survey_choice_id | course_survey_choice_id | course_survey_choice_id
+        //         1 |                  NULL |                    NULL |                NULL
+        //         2 |                     1 |                    NULL |                NULL
+        //         3 |                  NULL |                    NULL |                   1
+        //         4 |                  NULL |                       1 |                   3
+        //         5 |                     2 |                       2 |                NULL
+        //         6 |                     1 |                    NULL |                NULL
+        // in the above example
+        //  Survey with id 1 is returned for all Admins
+        //  Surveys with survey_id 1,2,6 are returned for all Instructors
+        //  Surveys with survey_id 1,2,5,6 are returned for all TAs
+
+        $sql =
+            "SELECT survey_id FROM surveys AS s " .
+            "WHERE $col_name IS NULL $cond AND s.term LIKE :term";
```

Comment:
Do we suddenly ignore the course code?

Reply:
I used course_code to query the user instead using course_code from user_associations.